### ISSUE TITLE
Feature matching loss

### ISSUE THREAD

**Issue by qiuxiaqing** [NVIDIA/pix2pixHD#89]
Dear sir:
        I don't know why you add "the feature matching loss".In your paper, you say that "We compare our final implementation to the results using (1) only GAN loss, and (2) GAN + feature matching loss (i.e., without VGG loss). The obtained preference rates are 68.55% and 58.90%, respectively." In this part, when you adding the feature matching loss, the preference is not good as GAN loss. 

        Please reply when it is convenient for you. Thank you very much!

**Comment by HarveyYesan** [comment]
Feature matching is presented in improved-GAN, which make the training more stable and is more information for generator。
However，I have two questions 
1) What's the meaning of parameter Ni in the equation(4) of paper ?
2) How to set up the weight in line 182 & 183 in pix2pixHD_model.py ?

Thanks!

**Comment by ianzen** [comment]
From my personal experience, feature matching loss causes training to be more stable and faster. But this is at the expense of the learned features to be not as deep as normal gan loss. The discriminator behaves akin to a reconstructive L1 loss you'd find in a vae, not really learning the distribution of the target image domain. This causes the generator to overfit for the training data and has difficulty generalizing to test data.
